Add "malicious maintainer" to the compromise definitions (#515)
cncf/sig-security: supply-chain-security/compromises/compromise-definitions.md

- [Wikipedia Code Signing](https://en.wikipedia.org/wiki/Code_signing)

## Malicious Maintainer

> Occurs when a maintainer deliberately injects a vulnerability somewhere in the supply chain or in the source code.

Reviewer: I think we want to capture people posing as maintainers (i.e. compromised a maintainer's account) in this definition too? Maybe

```suggestion
Occurs when a maintainer, or an entity posing as a maintainer, deliberately injects a vulnerability somewhere in the supply chain or in the source code.
```

Author: Makes sense.
Also, it could be hard to trace if the motive behind the maintainer actions, because even if the motives were bad initially, then
tomorrow he/she can easily say: "my account was hacked".

This kind of compromise could have great consequences because usually the individual executing the attack is considered trustworthy by many.

> This category includes attacks from experienced maintainers going rogue as well as new personas performing an attack soon after they have acquired responsibilities.

Reviewer: In-line with the proposal above:
```suggestion
This category includes attacks from experienced maintainers going rogue, maintainer account compromise, and new personas performing an attack soon after they have acquired responsibilities.
```
